pkgs/development/tools/extism-cli/default.nix

```
@@ -1,37 +1,29 @@
-{ lib, stdenvNoCC, fetchFromGitHub, python3, makeBinaryWrapper }:
+{
+  lib
+, buildGoModule
+, fetchFromGitHub
+}:
 
-stdenvNoCC.mkDerivation rec {
+buildGoModule rec {
   pname = "extism-cli";
-  version = "0.1.0";
-
+  version = "0.3.8";
   src = fetchFromGitHub {
     owner = "extism";
     repo = "cli";
-    rev = "97935786166e82154266b82410028482800e6061";
-    sha256 = "sha256-LRzXuZQt5h3exw43UXUwLVIhveYVFw/SQ2YtHI9ZnWc=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-StMipPMLSQzrhWv0yoKkNiuHMRW7QIhmVZ/M27WDWrM=";
   };
+  modRoot = "./extism";
 
-  buildInputs = [ python3 ];
-  nativeBuildInputs = [ makeBinaryWrapper ];
-
-  installPhase = ''
-    runHook preInstall
-
-    install -D -m 755 ./extism_cli/__init__.py "$out/bin/extism"
-
-    # The extism cli tries by default to install a library and header into /usr/local which does not work on NixOS.
-    # Pass a reasonable writable directory which can still be overwritten with another --prefix argument.
-    wrapProgram "$out/bin/extism" \
-      --add-flags '--prefix $HOME/.local'
+  vendorHash = "sha256-sSKiwYT5EP0FQJbhgv9ZFDwwwvIJ66yMULbj529AZwY=";
 
-    runHook postInstall
-  '';
+  doCheck = false; # Tests require network access
 
   meta = with lib; {
     description = "The extism CLI is used to manage Extism installations";
     homepage = "https://github.com/extism/cli";
     license = licenses.bsd3;
-    maintainers = with maintainers; [ ];
+    maintainers = with maintainers; [ zshipko ];
```

```suggestion
    mainProgram = "extism";
    maintainers = with maintainers; [ zshipko ];
```
mainProgram is a nice meta attribute allowing things like `lib.getExe` or `nix run` to find a sensible binary.